src/features/Dashboard/components/RecentTransactionList.tsx

```diff
@@ -34,62 +35,121 @@ import newWindowIcon from '@assets/images/icn-new-window.svg';
+import contractInteract from '@assets/images/transactions/contract-interact.svg';
+import contractDeploy from '@assets/images/transactions/contract-deploy.svg';
+import defizap from '@assets/images/transactions/defizap.svg';
+import membershipPurchase from '@assets/images/transactions/membership-purchase.svg';
+import swap from '@assets/images/transactions/swap.svg';
 
 interface Props {
   className?: string;
   accountsList: StoreAccount[];
 }
 
-enum ITxType {
+enum IStandardTxType {
   TRANSFER = 'TRANSFER',
   OUTBOUND = 'OUTBOUND',
   INBOUND = 'INBOUND'
 }
 
+const ITxHistoryType = { ...ITxType, ...IStandardTxType };
+type ITxHistoryType = Exclude<
+  Exclude<ITxType | IStandardTxType, ITxType.STANDARD>,
+  ITxType.UNKNOWN
+>;
+
+interface ITxHistoryEntry extends Omit<Omit<ITxReceipt, 'txType'>, 'timestamp'> {
+  network: Network;
+  toLabel: string;
+  fromLabel: string;
+  txType: ITxHistoryType;
+  timestamp: number;
+}
+
 interface ITxTypeConfigObj {
   icon: any;
   label(asset: Asset): string;
 }
 
 type ITxTypeConfig = {
-  [txType in ITxType]: ITxTypeConfigObj;
+  [txType in ITxHistoryType]: ITxTypeConfigObj;
 };
 
 const TxTypeConfig: ITxTypeConfig = {
-  [ITxType.INBOUND]: {
+  [ITxHistoryType.INBOUND]: {
     label: (asset: Asset) =>
       translateRaw('RECENT_TX_LIST_LABEL_RECEIVED', {
         $ticker: asset.ticker || translateRaw('UNKNOWN')
       }),
     icon: inbound
   },
-  [ITxType.OUTBOUND]: {
+  [ITxHistoryType.OUTBOUND]: {
     label: (asset: Asset) =>
       translateRaw('RECENT_TX_LIST_LABEL_SENT', {
         $ticker: asset.ticker || translateRaw('UNKNOWN')
       }),
     icon: outbound
   },
-  [ITxType.TRANSFER]: {
+  [ITxHistoryType.TRANSFER]: {
     label: (asset: Asset) =>
       translateRaw('RECENT_TX_LIST_LABEL_TRANSFERRED', {
         $ticker: asset.ticker || translateRaw('UNKNOWN')
       }),
     icon: transfer
+  },
+  [ITxHistoryType.DEFIZAP]: {
+    label: (asset: Asset) =>
+      translateRaw('RECENT_TX_LIST_LABEL_DEFIZAP_ADD', {
+        $ticker: asset.ticker || 'Unknown'
+      }),
+    icon: defizap
+  },
+  [ITxHistoryType.PURCHASE_MEMBERSHIP]: {
+    label: (_: Asset) => translateRaw('RECENT_TX_LIST_LABEL_MEMBERSHIP_PURCHASED'),
+    icon: membershipPurchase
+  },
+  [ITxHistoryType.SWAP]: {
+    label: (_: Asset) => translateRaw('RECENT_TX_LIST_LABEL_SWAP'),
+    icon: swap
+  },
+  [ITxHistoryType.APPROVAL]: {
+    label: (_: Asset) => translateRaw('RECENT_TX_LIST_LABEL_APPROVAL'),
+    icon: approval
+  },
+  [ITxHistoryType.CONTRACT_INTERACT]: {
+    label: (_: Asset) => translateRaw('RECENT_TX_LIST_LABEL_CONTRACT_INTERACT'),
+    icon: contractInteract
+  },
+  [ITxHistoryType.DEPLOY_CONTRACT]: {
+    label: (_: Asset) => translateRaw('RECENT_TX_LIST_LABEL_CONTRACT_DEPLOY'),
+    icon: contractDeploy
   }
 };
 
-export const deriveTxType = (accountsList: StoreAccount[], tx: ITxReceipt) => {
+export const deriveTxType = (accountsList: StoreAccount[], tx: ITxReceipt): ITxHistoryType => {
```

Comment:
Prefer simple presentational components.
Please create and test utility function to extract this businnes logic.

Reply:
I extracted this out to a helpers.ts file in the `src/features/Dashboard` folder. Still needs test